druid-shell/src/mac/application.rs

```diff
@@ -41,4 +43,44 @@ impl Application {
             msg_send![shared, hideOtherApplications];
         }
     }
+
+    /// Returns the contents of the clipboard, if any.
+    pub fn get_clipboard_contents() -> Option<ClipboardItem> {
+        unsafe {
+            let nspasteboard = class!(NSPasteboard);
+            let pasteboard: id = msg_send![nspasteboard, generalPasteboard];
+            let data_types: id = msg_send![pasteboard, types];
+            let count: usize = msg_send![data_types, count];
+
+            for i in 0..=count {
```

Comment:
This seems wrong, I believe we shouldn't include `count` in the iter. That said, tracing the code below, if we get a `nil`, `is_string` will be false for that, so I can see the code silently doing the right thing.